models/models.js

```diff
@@ -20,9 +20,24 @@ exports.receivedArticleById = (article_id) => {
     });
 };
 
-exports.fetchArticles = (sort_by, order) => {
+exports.fetchArticles = (sort_by, order, topic) => {
+  let queryArgs = [];
+  const promiseArray = [];
+  let queryCount = 0;
+
   let queryStr = `
-    SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id`;
+    SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id`;
+
+  if (topic) {
+    queryCount++;
+    queryStr += ` WHERE articles.topic = $${queryCount}`;
+    queryArgs.push(topic);
+    promiseArray.push(
+      db.query(`SELECT * FROM topics WHERE slug = $1`, [topic])
```

Comment:
This code works and is correctly checking the topic exists before the query is performed.
However, I think it could be a bit more readable. If you would like to refactor it, I'd potentially look at moving this into a utils file and calling it a function like `checkTopicExists`. You could then call this and as long as the function doesn't reject, the rest of the code could progress. I personally think this would be more readable than pushing to the Promises array, as it's quite hard to follow what's going on at the moment.
Completely up to you though! 😄